### ISSUE TITLE
[Feature] access run hostname through wandb.api

### ISSUE THREAD

**Issue by qsh-zh** [wandb/wandb#2305]
**Is your feature request related to a problem? Please describe.**
get details about one run

**Describe the solution you'd like**
```python
run = api.run("xxxx")
hostname = run.hostname
``` 

**Describe alternatives you've considered**

**Additional context**

**Comment by tyomhak** [comment]
Hey @qshzh , thank you for the feature request :)

**Comment by artem-tkachuk** [comment]
Quick "fix" or workaround while the official feature is still not here yet: set `wandb.run.tags = ["<HOSTNAME_HERE>"]` in your script (in my case I used `os.environ["SLURM_JOB_NODELIST"]` for that placeholder) and then get it with `run.tags[0]` when getting runs data through wandb API :)

**Comment by qsh-zh** [comment]
@artem-tkachuk I see. I did a similar thing by pushing a hostname into the configuration.

**Comment by iamhatesz** [comment]
Any chance for including this in API?

**Comment by dcx** [comment]
One more request for this feature! My use case: I'd like to know where to find the larger artifacts that don't get sent to wandb. Thanks!

**Comment by vanpelt** [comment]
@dcx what are you referring to when you say "larger artifacts that don't get sent to wandb"?

**Comment by dcx** [comment]
Apologies, that might not have been clear. So I'm doing a bit of research which involves training and evaluating a ton of variants of models on the same dataset. I'm using wandb sweeps for some of this. Each run produces large outputs such as model checkpoints and extra analyses (and so I don't send them to wandb; it's wasteful). Since there are so many runs, I spread them across a few machines, and run data is stored in folders named using Unix timestamps.

I was trying to write a script that read from wandb sweeps via the API, and pulled in the extra files for extra processing. But I realised I couldn't tell which machine the data for the runs in question were on, because timestamps are not unique.

I ended up solving this using the GUI - filtering by hostname, selecting all, attaching a tag for each machine, and reading those tags via scripting. And going forward I'm just adding the hostname to the run name. But this was a bit inconvenient; it was a little surprising that this wasn't already accessible. Hope that's helpful!

**Comment by vanpelt** [comment]
Thanks @dcx another option for now is to add a key to config.  So you could do `wandb.config.update({"hostname": "myhostname"})` then you can always access that value.  I'll re-open this ticket because we should be exposing the hostname.

**Comment by janosh** [comment]
This seems to be a workaround for now. Any update on 1st party support? Would be great if all of `wandb-metadata.json` was easier to access.

```py
import wandb

run = wandb.Api().run("<entity>/<project>/<run_id>")

meta = json.load(run.file("wandb-metadata.json").download(replace=True))

meta
>>> {'os': 'Linux-4.18.0-372.32.1.el8_6.x86_64-x86_64-with-glibc2.28',
 'python': '3.9.6',
 'heartbeatAt': '2022-11-22T04:24:32.521072',
 'startedAt': '2022-11-22T04:24:31.905318',
 'docker': None,
 'cuda': None,
 'args': [],
 'state': 'running',
 'program': '/rds/user/hpc-work/models/train.py',
 'codePath': 'models/train.py',
 'git': {'remote': 'https://github.com/***************',
  'commit': '9a8c146e417523b78a37ef300a159e1cb7dea767'},
 'email': '********************',
 'root': '/rds/user/hpc-work',
 'host': 'gpu-q-53',
 'username': ',
 'executable': '/rds/user/hpc-work/.venv/py39/bin/python',
 'cpu_count': 128,
 'cpu_count_logical': 128,
 'cpu_freq': {'current': 2470.236578125003, 'min': 0.0, 'max': 0.0},
 'cpu_freq_per_core': [{'current': 2445.534, 'min': 0.0, 'max': 0.0},
  {'current': 2445.534, 'min': 0.0, 'max': 0.0},
  ...
  {'current': 2445.534, 'min': 0.0, 'max': 0.0}],
 'disk': {'total': 1396.5678596496582, 'used': 20.321372985839844},
 'gpu': 'NVIDIA A100-SXM4-80GB',
 'gpu_count': 1,
 'gpu_devices': [{'name': 'NVIDIA A100-SXM4-80GB',
   'memory_total': 85899345920}],
 'memory': {'total': 1007.0768737792969}}

meta['host']
>>> 'gpu-q-53'
```

**Comment by thanos-wandb** [comment]
Hi @janosh @talesa @qsh-zh @tyomhak @artem-tkachuk @iamhatesz @dcx apologies for the lack of any update here, this slipped through the cracks. Our engineers in SDK team have released this functionality in wandb v0.15.10. You could use the API as follows to get the run's hostname:

```
import wandb
api = wandb.Api()
run = api.run("entity/project/run-id")
print(run.metadata["host"])
```

I hope this helps! I will now close this ticket on our end, but please let us know if you had any questions or issues.

**Comment by JesseSilverberg** [comment]
This snippet doesn't seem to work for me. I want to read the hostname of an existing run, and while the `run` object has the metadata key (and `_metadata`; https://github.com/wandb/wandb/pull/9043? ), they both seem to be empty:
```
>>> dir(run)
['__class__', '__delattr__', '__dict__', '__dir__', '__doc__', '__eq__', '__firstlineno__', '__format__', '__ge__', '__getattr__', '__getattribute__', '__getstate__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__sizeof__', '__static_attributes__', '__str__', '__subclasshook__', '__weakref__', '_attrs', '_base_dir', '_entity', '_exec', '_files', '_full_data_loaded', '_full_history', '_include_sweeps', '_is_loaded', '_lazy', '_load_from_attrs', '_load_with_fragment', '_metadata', '_project_internal_id', '_repr_html_', '_sampled_history', '_server_provides_project_id_field', '_state', '_summary', 'client', 'config', 'create', 'delete', 'dir', 'display', 'entity', 'file', 'files', 'history', 'id', 'json_config', 'lastHistoryStep', 'load', 'load_full_data', 'log_artifact', 'logged_artifacts', 'metadata', 'name', 'path', 'project', 'rawconfig', 'save', 'scan_history', 'server_provides_internal_id_field', 'snake_to_camel', 'state', 'storage_id', 'summary', 'summary_metrics', 'sweep', 'sweep_name', 'system_metrics', 'to_html', 'update', 'upload_file', 'url', 'use_artifact', 'used_artifacts', 'user', 'username', 'wait_until_finished']
>>> run.metadata
>>> run._metadata
```
